packages/flutter/lib/src/material/dialog.dart

```diff
@@ -1129,3 +1140,71 @@ double _paddingScaleFactor(double textScaleFactor) {
   // a non-scaled padding of 24 will produce a padding between 24 and 8.
   return lerpDouble(1.0, 1.0 / 3.0, clampedTextScaleFactor - 1.0)!;
 }
+
+// Generate a DialogTheme that represents the M2 default values.
+// This was generated by hand from the previous hand coded defaults
+// for M2. It uses get method overrides instead of properties to
+// avoid computing values that we may not need upfront.
+class _M2Defaults extends DialogTheme {
+  _M2Defaults(this.context)
+    : _textTheme = Theme.of(context).textTheme;
+
+  final BuildContext context;
+  final TextTheme _textTheme;
+
+  @override
+  AlignmentGeometry? get alignment =>  Alignment.center;
```

Comment:
For properties that don't need a computation (like looking up a color from the ColorScheme), you can just include them as a parameter to the super class in the constructor. The main reason we want to override some of the properties with getter methods is so we aren't computing something until it actually needed. In the case of constants we don't need to do this.